mm2src/coins/eth.rs

```diff
@@ -2628,42 +2629,61 @@ impl MarketCoinOps for EthCoin {
                 },
             };
 
-            let events = match self
-                .spend_events(swap_contract_address, args.from_block, current_block)
-                .compat()
-                .await
-            {
-                Ok(ev) => ev,
-                Err(e) => {
-                    error!("Error getting spend events: {}", e);
-                    Timer::sleep(5.).await;
-                    continue;
-                },
-            };
+            if tx_hash.is_none() {
+                let mut next_from_block = args.from_block;
 
-            let found = events.iter().find(|event| &event.data.0[..32] == id.as_slice());
+                // Split the range into windows of size logs_block_range
+                while next_from_block <= current_block {
+                    let to_block = std::cmp::min(next_from_block + self.logs_block_range - 1, current_block);
 
-            if let Some(event) = found {
-                if let Some(tx_hash) = event.transaction_hash {
-                    let transaction = match self.transaction(TransactionId::Hash(tx_hash)).await {
-                        Ok(Some(t)) => t,
-                        Ok(None) => {
-                            info!("Tx {} not found yet", tx_hash);
-                            Timer::sleep(args.check_every).await;
```

Comment:
it's confusing why we use 5s sleep and in other situations we use `check_every`. but i see you preserved the logic as it was :+1:

we can leave a todo about this (if there is no justification).
i think we should always do `check_every`, as that's the expected wait time when failure or no result found. using a constant like 5s might not play well with certain (`check_every` & `wait_until`) pair.

Reply:
> it's confusing why we use 5s sleep and in other situations we use check_every. but i see you preserved the logic as it was

yes, I decided to leave time sleep in legacy as it is, as Im not sure about the reason of using `check_every` and 5 sec in different places. When it comes to legacy Im trying to balance between "do the refactor when we need it" and "Does it work? Dont touch it". 

I used check_every everywhere in `find_taker_payment_spend_tx_impl` (its an new protocol alternative of legacy `wait_for_htlc_tx_spend`). So we can have a fresh view of this behaviour.